src/lib.rs

```diff
@@ -1,127 +1,734 @@
+    /// Create a bitset from the underlying bit block.
+    fn from(block: T) -> Self {
+        Self { blocks: [block] }
+    }
+}
+
+impl<T: BitBlock, const N: usize> Index<usize> for TinyBitSet<T, N> {
+    type Output = bool;
+
+    fn index(&self, index: usize) -> &Self::Output {
+        if (self.blocks[index / T::BITS] >> (index % T::BITS)) & T::LSB == T::LSB {
+            &true
+        } else {
+            &false
+        }
+    }
+}
+
+impl<T: BitBlock, const N: usize> Not for TinyBitSet<T, N> {
     type Output = Self;
 
+    /// Returns a bitset with all bits flipped.
+    fn not(self) -> Self::Output {
+        array::from_fn(|i| !self.blocks[i]).into()
+    }
+}
+
+impl<T: BitBlock, const N: usize> BitAnd for TinyBitSet<T, N> {
+    type Output = Self;
+
+    /// Returns a bitset with all bits that are set in both `self` and `rhs`.
     fn bitand(self, rhs: Self) -> Self::Output {
-        Self(array::from_fn(|i| self.0[i] & rhs.0[i]))
+        array::from_fn(|i| self.blocks[i] & rhs.blocks[i]).into()
     }
 }
 
-impl<T: Block, const N: usize> BitAndAssign for BitSet<T, N> {
+impl<T: BitBlock, const N: usize> BitAndAssign for TinyBitSet<T, N> {
     fn bitand_assign(&mut self, rhs: Self) {
         *self = *self & rhs;
     }
 }
 
-impl<T: Block, const N: usize> BitOr for BitSet<T, N> {
+impl<T: BitBlock, const N: usize> BitOr for TinyBitSet<T, N> {
     type Output = Self;
 
+    /// Returns a bitset with all bits that are set in either `self` or `rhs`.
     fn bitor(self, rhs: Self) -> Self::Output {
-        Self(array::from_fn(|i| self.0[i] | rhs.0[i]))
+        array::from_fn(|i| self.blocks[i] | rhs.blocks[i]).into()
     }
 }
 
-impl<T: Block, const N: usize> BitOrAssign for BitSet<T, N> {
+impl<T: BitBlock, const N: usize> BitOrAssign for TinyBitSet<T, N> {
     fn bitor_assign(&mut self, rhs: Self) {
         *self = *self | rhs;
     }
 }
 
-impl<T: Block, const N: usize> Not for BitSet<T, N> {
+impl<T: BitBlock, const N: usize> BitXor for TinyBitSet<T, N> {
     type Output = Self;
 
-    fn not(self) -> Self::Output {
-        Self(self.0.map(|block| !block))
+    /// Returns a bitset with all bits that are set in exactly one of `self` and `rhs`.
+    fn bitxor(self, rhs: Self) -> Self::Output {
+        array::from_fn(|i| self.blocks[i] ^ rhs.blocks[i]).into()
+    }
+}
+
+impl<T: BitBlock, const N: usize> BitXorAssign for TinyBitSet<T, N> {
+    fn bitxor_assign(&mut self, rhs: Self) {
+        *self = *self ^ rhs;
+    }
+}
+
+impl<T: BitBlock, const N: usize> Debug for TinyBitSet<T, N> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "TinyBitSet([")?;
+        for (i, block) in self.blocks.into_iter().enumerate() {
+            if i != 0 {
+                write!(f, ", ")?;
+            }
+
+            // The `+ 2` accounts for the `0b` prefix
+            write!(f, "{block:#0width$b}", width = T::BITS + 2)?;
+        }
+        write!(f, "])")
```

Comment:
I think this can be simplified by using 
```rust
f.debug_list().entries(iter).finish()
```
I'm not sure how to write the items efficiently in this case but there might be a way for it

Reply:
I we want to use `debug_list` I don't think theres a way around writing a newtype wrapping the blocks that customizes the `Debug` formatting. I will try it out at some point, but I don't think it will end up much cleaner.